File: backend/internal/model/portfolio.go
package model

// Portfolio is the top-level response served by the API.
// Customize owner info, tech stack, and projects in internal/data/portfolio.json.
type Portfolio struct {
	Owner            Owner             `json:"owner"`
	TechStack        []string          `json:"techStack"`
	FeaturedProjects []FeaturedProject `json:"featuredProjects"`
	ArchiveURL       string            `json:"archiveUrl"`
	SocialLinks      []SocialLink      `json:"socialLinks"`
}

// Owner holds personal branding shown in the hero/header area.
type Owner struct {
	Name  string `json:"name"`
	Title string `json:"title"`
	Bio   string `json:"bio"`
	Email string `json:"email"`
}

// FeaturedProject represents a card in the Featured Deployments section.
type FeaturedProject struct {
	ID          string   `json:"id"`
	Title       string   `json:"title"`
	Description string   `json:"description"`
	Tags        []string `json:"tags"`
	ImageURL    string   `json:"imageUrl"`
	ProjectURL  string   `json:"projectUrl"`
	RepoURL     string   `json:"repoUrl"`
}

// SocialLink is a footer/sidebar social profile link.
type SocialLink struct {
	Platform string `json:"platform"`
	URL      string `json:"url"`
	Label    string `json:"label"`
}

File: frontend/src/types/portfolio.ts
// Portfolio domain types shared across components and API layer.

export interface Project {
  id: string;
  title: string;
  description: string;
  imageUrl: string;
  tags: string[];
  projectUrl?: string;
  featured: boolean;
}

export interface ArchiveLink {
  label: string;
  url: string;
}

export interface ProjectsResponse {
  projects: Project[];
  archive: ArchiveLink;
}

export interface TechStackItem {
  name: string;
}

export interface TechStackResponse {
  items: TechStackItem[];
}

export interface Profile {
  name: string;
  title: string;
  tagline: string;
  email: string;
  github: string;
  linkedin?: string;
  location?: string;
  about: string;
  contactLabel: string;
}

export interface PortfolioData {
  profile: Profile;
  projects: ProjectsResponse;
  techStack: TechStackResponse;
}
